mitosheet/mitosheet/step_performers/step_performer.py

```diff
@@ -82,6 +91,101 @@ def transpile(
+        prev_state: State,
+        params: Dict[str, Any],
+        execution_data: Optional[Dict[str, Any]]=None,
+        new_dataframe_params: Optional[ExecuteThroughTranspileNewDataframeParams]=None,
+        renamed_column_headers: bool=False,
+        new_column_params: Optional[ExecuteThroughTranspileNewColumnParams]=None,
+        use_deprecated_id_algorithm: Optional[bool]=False
+    ) -> Tuple[State, Dict[str, Any]]:
+        """
+        Some steps can be executed through the transpiled code -- and in these cases, we can call this function
+        so that we don't have to duplicate the work we do
+        """
+        if execution_data is None:
+            execution_data = {}
+
+        modified_dataframe_indexes = cls.get_modified_dataframe_indexes(params)
+        # If the modified indexes are -1, then only new dataframes have been created -- and in this
+        # case we just don't detect modifications
+        if modified_dataframe_indexes == {-1}:
+            modified_dataframe_indexes = set()
+
+        post_state = prev_state.copy(deep_sheet_indexes=modified_dataframe_indexes)
+
+        code_chunks = cls.transpile(post_state, params, execution_data)
+        code = []
+        for chunk in code_chunks:
+            _code, imports = chunk.get_code()
+            code.extend(imports)
+            code.extend(_code)
+
+        final_code = "\n".join(code)
+
+        # NOTE; this is all wrong, but it's just b/c it's hard to get things out of exec. We do our best
+        exec_globals = get_globals_for_exec(post_state, post_state.public_interface_version)
+        exec_locals = {**exec_globals}
+        
+        pandas_start_time = perf_counter()
+        exec(final_code, exec_globals, exec_locals)
+        pandas_processing_time = perf_counter() - pandas_start_time
+
+
+        for modified_dataframe_index in modified_dataframe_indexes:
+            df_name = prev_state.df_names[modified_dataframe_index]
+            new_df = exec_locals[df_name]
+            old_df = prev_state.dfs[modified_dataframe_index]
+            post_state.dfs[modified_dataframe_index] = new_df
+
+            if renamed_column_headers:
+                # If we renamed column headers, update them in the state
+                for old_header, new_header in zip(old_df.columns, new_df.columns):
+                    column_id = prev_state.column_ids.get_column_id_by_header(modified_dataframe_index, old_header)
+                    post_state.column_ids.set_column_header(modified_dataframe_index, column_id, new_header)
+
+            if len(new_df.columns) > len(old_df.columns):
+                # Find all the new column headers, and add them to the state
+                new_column_headers = [ch for ch in new_df.columns if ch not in old_df]
+                post_state.add_columns_to_state(modified_dataframe_index, new_column_headers)
+
+                # Check if there are any invalid column headers
+                c = Counter(new_df.columns)
```

Comment:
Can we turn this column header checking into a utility and add a doc string?

Reply:
It's simple enough,let's keep it as is for now.

Reply:
Jk, you're right, this function needs cleaning.